### PR TITLE
added priority field

### LINKED ISSUE #37: Add a priority field
### Elevator pitch: Priority field
Allows tasks to be asigned to a certain priority (Low, Medium, High).

### Describe how this new feature adds value to the application
By having priorities, users can easily see which tasks are more important than others so they can decide which ones to work on first.

### Describe suggested implementation
We could edit the parsing grammar for add command to make it similar to edit command in #35 such as: `add [description] [priority]`. It could even be a single letter like H for high, M for medium and L for low.

### PR DESCRIPTION
**Pull request by Barraza44**
Closes #37 
Adds a priority field for tasks

### CODE REVIEW

**Comment by pobiega** [comment]
So far so good, except you made priority a mandatory field. Would be better if you made it optional.

File: BusyList/Commands/Commands.cs [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
     public record ReadCommand(int Id) : Command;
     public record NextCommand() : Command;
-    public record AddCommand(string Description) : Command;
+    public record AddCommand(string Description, string Priority) : Command;
```
Comment (pobiega) [inline BusyList/Commands/Commands.cs:7]: Priority is not optional here.

File: BusyList/Handlers/NextHandler.cs [hunk 2]
```diff
@@ -25,9 +25,13 @@ public void Run(NextCommand command)
             {
                 Console.WriteLine($"Task id: {item.Id}");
```
Comment (pobiega) [inline BusyList/Handlers/NextHandler.cs:26]: Shouldn't this entire thing use `item.Print()`?

File: BusyList/TaskItem.cs [hunk 3]
```diff
@@ -4,24 +4,31 @@ namespace BusyList
 {
     public class TaskItem
     {
-        public TaskItem(int id, string description, TaskStatus status = TaskStatus.NotStarted)
+        public TaskItem(int id, string description, string priority, TaskStatus status = TaskStatus.NotStarted)
```
Comment (pobiega) [inline BusyList/TaskItem.cs:7]: Is `string` the best data type to store priority in? What alternatives can you think of?

**Review by pobiega — CHANGES_REQUESTED** [review]

File: BusyList/Parsing/CommandGrammar.cs [hunk 4]
```diff
@@ -23,6 +24,15 @@ public static class CommandGrammar
         private static readonly Parser<int> _number =
             Parse.Number.Select(s => int.Parse(s));
 
+        private static readonly Parser<string> _priority =
```
Comment (pobiega) [inline BusyList/Parsing/CommandGrammar.cs:27]: While this certainly works, see if you can't use [the Sprache Return command](https://justinpealing.me.uk/post/2020-04-06-sprache5-select-and-return/#return) to clean this up.

Comment (pobiega) [reply BusyList/Parsing/CommandGrammar.cs:27]: If you need help, reach out to me on discord and we'll figure it out.

File: BusyList/PriorityEnum.cs [hunk 5]
```diff
@@ -0,0 +1,28 @@
+﻿namespace BusyList
+{
+    public enum PriorityEnum
```
Comment (pobiega) [inline BusyList/PriorityEnum.cs:3]: Good use of an enum.

**Review by pobiega — CHANGES_REQUESTED** [review]

**Review by pobiega — DISMISSED** [review]
Looks really good, great work!

File: BusyList/Commands/Commands.cs [hunk 6]
```diff
@@ -4,7 +4,7 @@
 
     public record ReadCommand(int Id) : Command;
     public record NextCommand() : Command;
-    public record AddCommand(string Description) : Command;
+    public record AddCommand(string Description, PriorityEnum? Priority = null) : Command;
```
Comment (chrisK00) [inline BusyList/Commands/Commands.cs:7]: Shouldnt the default be Low - considering its low prioritized. A todo will always have a priority, and if we simply dont care it means its low prioritized

**Review by chrisK00 — DISMISSED** [review]
Well done building the parser!

File: BusyList/Commands/Commands.cs (hunk 6 above)
Comment (pobiega) [reply BusyList/Commands/Commands.cs:7]: Interesting take. If we go with "all items are prioritized" then I suggest we rename "medium" to "normal" and have that be our default. It makes sense to be able to make tasks that are explicitly lower or higher than the default.

Comment (Barraza44) [reply BusyList/Commands/Commands.cs:7]: So, should I change it? 😁

Comment (unbekanntunity) [reply BusyList/Commands/Commands.cs:7]: If u want

Comment (pobiega) [reply BusyList/Commands/Commands.cs:7]: As the guy who both suggested and implemented the feature, I think its up to you to decide. If we decide to have "no priority" be the default, how will it compare to a low, a medium and a high one?

**Review by pobiega — APPROVED** [review]

**Review by chrisK00 — APPROVED** [review]

**Pull request merged**